app/upgrade_network_config.go

```diff
@@ -549,3 +550,131 @@ func (c *ProdDevContract) GetContracts(contracts []string) []string {
 
 	return contracts
 }
+
+func verifyAddress(address string, expectedPrefix *string) error {
+	prefix, decodedAddrData, err := bech32.DecodeAndConvert(address)
+	if err != nil {
+		return err
+	}
+	if expectedPrefix != nil && prefix != *expectedPrefix {
+		return fmt.Errorf("expected address prefix %s, got %s", *expectedPrefix, prefix)
+	}
+
+	reconstructedAddr, err := bech32.ConvertAndEncode(prefix, decodedAddrData)
+	if err != nil {
+		return err
+	}
+	if address != reconstructedAddr {
```

Comment:
This effectively ensures, that the original `address` value does not contain preceding and trailing whitespace characters.